Libraries/libutil/Sources/DefaultFilesystem.cpp

```diff
@@ -178,6 +145,55 @@ write(std::vector<uint8_t> const &contents, std::string const &path)
     return true;
 }
 
+bool DefaultFilesystem::
+copyFile(std::string const &from, std::string const &to)
+{
+#if defined(__APPLE__) || defined(__FreeBSD__)
+    if (!this->isFile(from)) {
+        return false;
+    }
+
+    if (this->isFile(to)) {
+        if (!this->removeFile(to)) {
+            return false;
+        }
+    }
+
+    copyfile_state_t state = ::copyfile_state_alloc();
+    copyfile_flags_t flags = COPYFILE_ALL | COPYFILE_NOFOLLOW;
+    if (::copyfile(from.c_str(), to.c_str(), state, flags)) {
+        return false;
+    }
+    ::copyfile_state_free(state);
+
+    return true;
+#else
+    return Filesystem::copyFile(from, to);
+#endif
+}
+
+bool DefaultFilesystem::
+removeFile(std::string const &path)
+{
+    if (::unlink(path.c_str()) < 0) {
+        if (::unlink(path.c_str()) < 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool DefaultFilesystem::
+isSymbolicLink(std::string const &path) const
+{
+    struct stat st;
+    if (::lstat(path.c_str(), &st) < 0) {
```

Comment:
Style curiosity question:  any qualms to shortening this down with something like

  return (::lstat(path.c_str(), &st) == 0) && S_ISLNK(st.st_mode);